### PR TITLE
add a note re selectorless services and proxy

### PR DESCRIPTION
**Pull request by khenidak**
Solves https://github.com/kubernetes/kubernetes/issues/105092

Adds a note re selectorless services and api-server proxy.

### CODE REVIEW

**Comment by khenidak** [comment]
cc @thockin  and @aojea

File: content/en/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -183,6 +183,13 @@ Accessing a Service without a selector works the same as if it had a selector.
 In the example above, traffic is routed to the single endpoint defined in
 the YAML: `192.0.2.42:9376` (TCP).
 
+{{< note >}}
+api-server does not allow proxying to endpoints that are not mapped to pods. Actions such as
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:187]: nit:
```suggestion
The Kubernetes API server does not allow proxying to endpoints that are not mapped to pods. Actions such as
```
or, simplifying the logic:
```suggestion
The Kubernetes API server only allow proxyings to endpoints that map to pods. Actions such as
```
We'd write “API server” or kube-apiserver, but not api-server.

File: content/en/docs/concepts/services-networking/service.md [hunk 2]
```diff
@@ -183,6 +183,13 @@ Accessing a Service without a selector works the same as if it had a selector.
...
+`kubectl proxy <service-name>` where the service is without a selector will fail due to this 
+constraint. This is due to security constraint where api-server can be used to proxy to external
+endpoints where the caller may not have access to but api-server does.
```
Comment (sftim) [inline content/en/docs/concepts/services-networking/service.md:190]:
```suggestion
constraint.

This restriction avoids the risk of attempted privilege escalation, where a user tries
to proxy via the API server to reach endpoints where the caller may not have access but the API
server does. 
```
?

**Review by sftim — COMMENTED** [review]
I think with tech review this could merge as-is. I made some suggestions aimed at improving the wording.

/sig api-machinery
(is that right?)

**Comment by aojea** [comment]
don't we have the CVE?

**Comment by khenidak** [comment]
@aojea  there was no CVE. But this is the ref:https://github.com/kubernetes/kubernetes/issues/58761

File: content/en/docs/concepts/services-networking/service.md [hunk 3]
```diff
@@ -183,6 +183,13 @@ Accessing a Service without a selector works the same as if it had a selector.
 In the example above, traffic is routed to the single endpoint defined in
 the YAML: `192.0.2.42:9376` (TCP).
 
+{{< note >}}
+The Kubernetes API server does not allow proxying to endpoints that are not mapped to pods. Actions such as
+`kubectl proxy <service-name>` where the service is without a selector will fail due to this 
+constraint. This is due to security constraint where api-server can be used to proxy to external
+endpoints where the caller may not have access to but api-server does.
```
Comment (aojea) [inline content/en/docs/concepts/services-networking/service.md:190]: you only replaced one api-server by `Kubernetes API server` , this lines still have `api-server does`

**Comment by khenidak** [comment]
@sftim can we move this forward?

File: content/en/docs/concepts/services-networking/service.md [hunk 4]
```diff
@@ -183,6 +183,14 @@ Accessing a Service without a selector works the same as if it had a selector.
 In the example above, traffic is routed to the single endpoint defined in
 the YAML: `192.0.2.42:9376` (TCP).
 
+{{< note >}}
+The Kubernetes API server does not allow proxying to endpoints that are not mapped to 
+pods. Actions such as `kubectl proxy <service-name>` where the service is without a
```
Comment (thockin) [inline content/en/docs/concepts/services-networking/service.md:188]: s/is without a/has no/

File: content/en/docs/concepts/services-networking/service.md [hunk 5]
```diff
@@ -183,6 +183,14 @@ Accessing a Service without a selector works the same as if it had a selector.
 In the example above, traffic is routed to the single endpoint defined in
 the YAML: `192.0.2.42:9376` (TCP).
 
+{{< note >}}
+The Kubernetes API server does not allow proxying to endpoints that are not mapped to 
+pods. Actions such as `kubectl proxy <service-name>` where the service is without a 
+selector will fail due to this constraint. This is due to security constraint where
```
Comment (thockin) [inline content/en/docs/concepts/services-networking/service.md:189]: I'd say "This prevents the Kubernetes API server from being used as a proxy to endpoints the caller may not be authorized to access."

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by khenidak** [comment]
@sftim i think this is ready now. PTAL.

**Pull request merged**